assembly/__tests__/ScaleString.spec.ts

```diff
@@ -0,0 +1,49 @@
+import { ScaleString } from "../ScaleString";
+
+describe("String", () => {
+
+    it("should encode string", () => {
+        const test1 = new ScaleString("go-pre");
+        expect<Array<u8>>(test1.toU8a()).toStrictEqual(append([0x18], "go-pre"));
```

Comment:
I think that a better test is to check the exact `byte` array that must come out of this encoding. Instead of appending "go-pre", I think that it would be better to hardcoded the expected byte array